syntax = "proto3";
package csce438;

// ------------------------------------------------------------
// The coordinator service definition
// These are the different functionalities exposed by service
// ------------------------------------------------------------

//Init and Heartbeat potentially redundant
service CoordService{
    rpc Heartbeat (ServerInfo) returns (Confirmation) {}
    rpc GetServer (ID) returns (ServerInfo) {}

    // 修改 Status 为 CoordStatus，避免 gRPC 命名冲突
    rpc create (PathAndData) returns (CoordStatus) {}
    rpc exists (Path) returns (CoordStatus) {}
}

// server info message definition
message ServerInfo{
    int32 serverID = 1;
    string hostname = 2;
    string port = 3;
    string type = 4;
}

// confirmation message definition
message Confirmation{
    bool status = 1;
}

// id message definition
message ID{
    int32 id = 1;
}

// ServerList definition
message ServerList{
    repeated string serverList = 1; 
}

// SynchService definition
message SynchService{
    string test = 1;
}

// PathAndData definition for rpc create
message PathAndData{
    string path = 1;
    string data = 2;
}

// path definition for rpc exists
message Path{
    string path = 1;
}

// 修改 Status 为 CoordStatus
message CoordStatus {
    bool status = 1;
}
